contract/contracts/Subscriptions.sol

```diff
@@ -193,6 +195,26 @@ contract Subscriptions is Ownable {
         emit Extend(user, end);
     }
 
+    /// @param user Owner for the pending subscription.
+    /// @param start Start timestamp for the pending subscription.
+    /// @param end End timestamp for the pending subscription.
+    /// @param rate Rate for the pending subscription.
+    /// @notice Creates a subscription template without requiring funds. Expected to be used with
+    /// `fulfil`.
+    function setPendingSubscription(address user, uint64 start, uint64 end, uint128 rate) public {
+        pendingSubscriptions[user] = Subscription({ start: start, end: end, rate: rate });
+    }
+
+    /// @param _to Owner of the new subscription.
+    /// @notice Equivalent to calling `subscribe` with the previous `setPendingSubscription`
+    /// arguments for the same user.
+    /// @dev Second param required, but currently unused.
+    function fulfil(address _to, uint256) public {
+        Subscription memory sub = pendingSubscriptions[_to];
```

Comment:
Do we need to validate that the pending sub exists? Also, are there any security concerns with this method need to address? Is it problematic if I call `fulfil` for someone else's subscription? Might be. Just want to bring it up

Reply:
It might be too restrictive, but can we say only the address that created the pending sub can call `fulfil` on it?

Reply:
Unfortunately, we have to assume that the caller is not the user who created the pending sub. It's probably fine that anyone could fulfil for the pending sub, since the sub itself is constructed by the user its for.